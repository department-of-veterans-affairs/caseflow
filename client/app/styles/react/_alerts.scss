
.usa-alert {

  &.fixed {
    position: fixed;
    top: 0;
    margin-top: 0;
    z-index: 999;
    left: 50%;
    transform: translate(-50%, 0);
  }

  // Remove after USWDS V.2.0
  &.usa-alert-slim {
    background-size: 4rem;

    .usa-alert-body {
      padding-left: 3.5rem;
    }

    .usa-alert-text {
      font-size: 2rem;
    }
  }

  .usa-alert-text {
    line-height: 1.5;
  }
}
