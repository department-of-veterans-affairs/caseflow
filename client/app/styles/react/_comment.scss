.comment-control-button-container {
  display: flex;
  width: 100%;
  padding: 3px 0;
  justify-content: space-between;
}

.comment-save-button-container {
  display: inline-block;
  width: 100%;
  margin-top: 5px;
  margin-bottom: 17px;
}

.comment-control-button {
  padding: 0;

  &.horizontal {
    text-align: left;
  }
}

.comment-control-button-divider {
  padding: 0 10px;
}

.comment-container-selected {
  box-shadow: 3px 3px 5px $color-gray-light;
}

.comment-container {
  word-wrap: break-word;
  word-break: break-word;
  background-color: $color-white;
  padding: 15px;
  cursor: pointer;

  &:not(.comment-horizontal-container) {
    margin-bottom: 15px;
    border-radius: 3px;
    border: 1px solid $color-gray-lighter;
  }

  &.comment-horizontal-container {
    padding: 25px 25px 25px 40px;
  }
}

// Max width is enforced so the size of this comment box
// fits with existing textareas in style guide
.comment-size-container {
  max-width: 430px;

  .cf-form-textarea {
    margin-bottom: 20px;
  }
}

.comment-horizontal-container {
  display: inline-block;
  width: 100%;
}

.comment-horizontal-spacing {
  margin-right: 49px;
}

.commentIcon-container {
  position: absolute;
  z-index: 10;
  // These dimensions need to match the comment icon SVG dimensions.
  width: 40px;
  height: 40px;
}

