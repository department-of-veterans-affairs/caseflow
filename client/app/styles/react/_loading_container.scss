.loadingContainer-positioning {
  position: absolute;
  width: 100%;
  height: 100%;
}

.loadingContainer-table {
  width: 100%;
  height: 100%;
  display: table;
}

.loadingContainer-table-cell {
  text-align: center;
  display: table-cell;
  vertical-align: middle;
}

.loadingContainer-content {
  position: relative;
  z-index: 1;

  iframe {
    width: 100%;
    height: 80vh;

    &.form9Viewer {
      height: 500px;
    }
  }
}

.loadingContainer-container {
  position: relative;
  overflow: hidden;
}

.cf-loading-button-text {
  display: inline-block;
  margin-right: 10px;
  line-height: 0;
  position: relative;
}

.cf-loading-message {
  .loading-message-text {
    display: inline-block;
    font-style: italic;
    font-weight: bolder;
  }

}

.cf-loading-button-symbol {
  display: inline-block;
}
