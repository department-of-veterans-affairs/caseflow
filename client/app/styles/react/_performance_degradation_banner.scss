.usa-banner {
  background-color: $color-gold-lightest;
  text-align: center;
}

.usa-banner-inner {
  padding: .8em;
}

.banner-icon {
  display: inline-block;
  vertical-align: text-top;
  // height: 25px;
}

.banner-text {
  padding-left: 6px;
}
