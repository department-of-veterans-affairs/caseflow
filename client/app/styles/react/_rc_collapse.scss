@mixin common() {

  width: 0;
  height: 0;
  font-size: 0;
  line-height: 0;
}

@mixin right($w, $h, $color) {

  border-top: $w solid transparent;
  border-bottom: $w solid transparent;
  border-left: $h solid $color;
}

@mixin bottom($w, $h, $color) {

  border-left: $w solid transparent;
  border-right: $w solid transparent;
  border-top: $h solid $color;
}

// scss-lint:disable IdSelector
#arrow {
  @include common();
  @include right(3px, 4px, $color-gray);
  @include bottom(3px, 4px, $color-gray);
}

.rc-collapse {

  &-anim-active {
    transition: height .2s ease-out;
  }

  & > &-item {
    &:first-child {
      border-top: none;
    }
  }

  & > &-item-disabled > .rc-collapse-header {
    cursor: not-allowed;
    color: $color-gray-light;
    background-color: $color-gray-lighter;
  }

  &-content {
    overflow: hidden;

    &-inactive {
      display: none;
    }
  }

  &-item:last-child {
    > .rc-collapse-content {
      border-radius: 0 0 3px 3px;
    }
  }

  & > &-item-active {
    > .rc-collapse-header {

      .arrow {

        #arrow > {
          @include bottom(3px, 4px, $color-gray);
        }
      }
    }
  }
}

.usa-accordion-content {
  &:not([aria-hidden]) {
    position: relative;
    left: 0;
  }
}
