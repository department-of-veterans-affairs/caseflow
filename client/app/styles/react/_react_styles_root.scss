@import '_comment.scss';
@import '_search_bar.scss';
@import '_performance_degradation_banner.scss';
@import '_rc_collapse.scss';
@import '_loading_container.scss';
@import '_select-dropdown.scss';
@import '_toggle_button.scss';
