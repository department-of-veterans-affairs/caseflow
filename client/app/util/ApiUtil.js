import request from 'superagent';
import nocache from 'superagent-no-cache';
import ReactOnRails from 'react-on-rails';

// TODO(jd): Fill in other HTTP methods as needed
const ApiUtil = {

  // Converts regular language to camelCase
  // 'VACOLS - 123, User' becomes 'vacolsUser'
  convertToCamelCase(phrase = '') {
    // Code courtesy of Stack Overflow, Question 2970525
    return phrase.toLowerCase().
        replace(/[^a-zA-Z ]/g, "").
        replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, (match, index) => {
          if (Number(match) === 0) {
            return "";
          }

          return index === 0 ? match.toLowerCase() : match.toUpperCase();
        });
  },

  // Converts camelCase to snake_case
  convertToSnakeCase(data = {}) {
    let result = {};

    for (let key in data) {
      if ({}.hasOwnProperty.call(data, key)) {
        // convert key from camelCase to snake_case
        let snakeKey = key.replace(/([A-Z])/g, ($1) => `_${$1.toLowerCase()}`);

        // assign value to new object
        result[snakeKey] = data[key];
      }
    }

    return result;
  },

  get(url, options = {}) {
    return request.
      get(url).
      set(this.headers(options.headers)).
      query(options.query).
      use(nocache);
  },

  // Default headers needed to talk with rails server.
  // Including rail's authenticity token
  headers(options = {}) {
    let headers = Object.assign({},
      {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      },
      ReactOnRails.authenticityHeaders(),
      options);


    return headers;
  },

  patch(url, options = {}) {
    return request.
      post(url).
      set(this.headers({ 'X-HTTP-METHOD-OVERRIDE': 'patch' })).
      send(options.data).
      use(nocache);
  },

  post(url, options = {}) {
    return request.
      post(url).
      set(this.headers(options.headers)).
      send(options.data).
      use(nocache);
  }
};

export default ApiUtil;
