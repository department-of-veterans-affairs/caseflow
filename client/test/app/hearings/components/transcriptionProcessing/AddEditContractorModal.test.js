
import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import COPY from '../../../../../COPY';
import ApiUtil from '../../../../../app/util/ApiUtil';
import { sprintf } from 'sprintf-js';

import {
  clickSubmissionButton,
  enterTextFieldOptions,
} from '../../../queue/components/modalUtils';

import { AddEditContractorModal } from
  '../../../../../app/hearings/components/transcriptionProcessing/AddEditContractorModal';

const onCancel = jest.fn().mockImplementation(() => {
  'Cancel';
});

const onConfirm = jest.fn().mockImplementation(() => {
  'Confirm';
});

const testContractor = {
  id: 1,
  directory: 'box_directory',
  email: 'test@va.gov',
  name: 'New Contractor',
  phone: 'phone-number',
  poc: 'person-of-contact',
  current_goal: 1
};

const renderAddContractorModal = () => {
  return render(
    <AddEditContractorModal onCancel={onCancel} onConfirm={onConfirm} title="Add contractor" />
  );
};

beforeEach(() => {
  jest.mock('../../../../../app/util/ApiUtil');
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('Add contractor form', () => {
  test('contains expected labels, fields and buttons', () => {
    const component = renderAddContractorModal();

    expect(component).toMatchSnapshot();
  });

  test('calls onCancel when cancel button is clicked', async () => {
    renderAddContractorModal();
    clickSubmissionButton(COPY.TRANSCRIPTION_SETTINGS_CANCEL);
    expect(onCancel).toHaveBeenCalled();
  });

  test('returns an alert and a new contractor on successful submit', async () => {
    ApiUtil.post = jest.fn().mockResolvedValue({ body: { transcription_contractor: testContractor } });

    renderAddContractorModal();

    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_NAME, testContractor.name);
    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_DIRECTORY, testContractor.directory);
    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_POC, testContractor.poc);
    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_EMAIL, testContractor.email);
    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_PHONE, testContractor.phone);
    clickSubmissionButton(COPY.TRANSCRIPTION_SETTINGS_ADD);

    expect(ApiUtil.post).toHaveBeenCalledWith(
      '/hearings/find_by_contractor',
      { data: {
        transcription_contractor: {
          id: '',
          directory: testContractor.directory,
          email: testContractor.email,
          name: testContractor.name,
          phone: testContractor.phone,
          poc: testContractor.poc
        }
      } }
    );

    await waitFor(() => expect(onConfirm).toHaveBeenCalledWith({
      alert: {
        title: sprintf(COPY.TRANSCRIPTION_SETTINGS_CREATE_MESSAGE, testContractor.name),
        message: '',
        type: 'success'
      },
      transcription_contractor: testContractor
    }));

  });

  test('displays an error from the server on failed form submit', async () => {
    ApiUtil.post = jest.fn().mockRejectedValue();

    renderAddContractorModal();

    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_NAME, testContractor.name);
    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_DIRECTORY, testContractor.directory);
    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_POC, testContractor.poc);
    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_EMAIL, testContractor.email);
    enterTextFieldOptions(COPY.TRANSCRIPTION_SETTINGS_LABEL_PHONE, testContractor.phone);
    clickSubmissionButton(COPY.TRANSCRIPTION_SETTINGS_ADD);

    await waitFor(() =>
      expect(screen.getByText(COPY.TRANSCRIPTION_SETTINGS_ERROR_MESSAGE)).toBeInTheDocument()
    );
  });
});
