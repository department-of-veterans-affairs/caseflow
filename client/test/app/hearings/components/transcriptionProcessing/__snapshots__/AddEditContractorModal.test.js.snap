// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add contractor form contains expected labels, fields and buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        aria-describedby="modal_id-desc"
        aria-labelledby="modal_id-title"
        aria-modal="true"
        class="cf-modal active "
        id="modal_id"
        role="dialog"
      >
        <div
          class="cf-modal-body"
          id="custom-contractor-modal"
        >
          <button
            class="cf-modal-close"
            id="Add-contractor-button-id-close"
            type="button"
          >
            <span
              class="usa-sr-only"
            >
              Close
            </span>
            <svg
              class="cf-icon-close"
              height="55"
              viewBox="0 0 55 55"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                close
              </title>
              <g
                fill="#212121"
              >
                <path
                  d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
                />
              </g>
            </svg>
          </button>
          <div
            style="display: flex;"
          >
            <div
              data-css-1gs0ko2=""
            >
              <h1
                id="modal_id-title"
              >
                Add contractor
              </h1>
              <div
                data-css-o98ubo=""
              >
                <p>
                  This will add a contractor to the list of assignable contractors.
                </p>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="name"
                  >
                    <span>
                      Contractor name
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="name"
                      name="name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="directory"
                  >
                    <span>
                      Contractor folder name in box.com
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="directory"
                      name="directory"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="poc"
                  >
                    <span>
                      POC name
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="poc"
                      name="poc"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="phone"
                  >
                    <span>
                      POC phone number
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="phone"
                      name="phone"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="email"
                  >
                    <span>
                      POC email
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="email"
                      name="email"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="cf-modal-divider"
          />
          <div
            class="cf-modal-controls"
          >
            <span>
              <button
                class="cf-modal-link cf-btn-link cf-push-left usa-button"
                id="Add-contractor-button-id-0"
                type="button"
              >
                Cancel
              </button>
            </span>
            <span>
              <button
                class="cf-push-right usa-button-disabled usa-button"
                disabled=""
                id="Add-contractor-button-id-1"
                type="button"
              >
                Add contractor
              </button>
            </span>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      aria-describedby="modal_id-desc"
      aria-labelledby="modal_id-title"
      aria-modal="true"
      class="cf-modal active "
      id="modal_id"
      role="dialog"
    >
      <div
        class="cf-modal-body"
        id="custom-contractor-modal"
      >
        <button
          class="cf-modal-close"
          id="Add-contractor-button-id-close"
          type="button"
        >
          <span
            class="usa-sr-only"
          >
            Close
          </span>
          <svg
            class="cf-icon-close"
            height="55"
            viewBox="0 0 55 55"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              close
            </title>
            <g
              fill="#212121"
            >
              <path
                d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
              />
            </g>
          </svg>
        </button>
        <div
          style="display: flex;"
        >
          <div
            data-css-1gs0ko2=""
          >
            <h1
              id="modal_id-title"
            >
              Add contractor
            </h1>
            <div
              data-css-o98ubo=""
            >
              <p>
                This will add a contractor to the list of assignable contractors.
              </p>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="name"
                >
                  <span>
                    Contractor name
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="name"
                    name="name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="directory"
                >
                  <span>
                    Contractor folder name in box.com
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="directory"
                    name="directory"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="poc"
                >
                  <span>
                    POC name
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="poc"
                    name="poc"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="phone"
                >
                  <span>
                    POC phone number
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="phone"
                    name="phone"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="email"
                >
                  <span>
                    POC email
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="email"
                    name="email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="cf-modal-divider"
        />
        <div
          class="cf-modal-controls"
        >
          <span>
            <button
              class="cf-modal-link cf-btn-link cf-push-left usa-button"
              id="Add-contractor-button-id-0"
              type="button"
            >
              Cancel
            </button>
          </span>
          <span>
            <button
              class="cf-push-right usa-button-disabled usa-button"
              disabled=""
              id="Add-contractor-button-id-1"
              type="button"
            >
              Add contractor
            </button>
          </span>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Edit contractor form contains expected labels, fields and buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        aria-describedby="modal_id-desc"
        aria-labelledby="modal_id-title"
        aria-modal="true"
        class="cf-modal active "
        id="modal_id"
        role="dialog"
      >
        <div
          class="cf-modal-body"
          id="custom-contractor-modal"
        >
          <button
            class="cf-modal-close"
            id="Add-contractor-button-id-close"
            type="button"
          >
            <span
              class="usa-sr-only"
            >
              Close
            </span>
            <svg
              class="cf-icon-close"
              height="55"
              viewBox="0 0 55 55"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                close
              </title>
              <g
                fill="#212121"
              >
                <path
                  d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
                />
              </g>
            </svg>
          </button>
          <div
            style="display: flex;"
          >
            <div
              data-css-1gs0ko2=""
            >
              <h1
                id="modal_id-title"
              >
                Add contractor
              </h1>
              <div
                data-css-o98ubo=""
              >
                <p>
                  This will add a contractor to the list of assignable contractors.
                </p>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="name"
                  >
                    <span>
                      Contractor name
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="name"
                      name="name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="directory"
                  >
                    <span>
                      Contractor folder name in box.com
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="directory"
                      name="directory"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="poc"
                  >
                    <span>
                      POC name
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="poc"
                      name="poc"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="phone"
                  >
                    <span>
                      POC phone number
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="phone"
                      name="phone"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="email"
                  >
                    <span>
                      POC email
                    </span>
                  </label>
                  <div
                    class="input-container"
                  >
                    <input
                      class="cf-form-textinput"
                      id="email"
                      name="email"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="cf-modal-divider"
          />
          <div
            class="cf-modal-controls"
          >
            <span>
              <button
                class="cf-modal-link cf-btn-link cf-push-left usa-button"
                id="Add-contractor-button-id-0"
                type="button"
              >
                Cancel
              </button>
            </span>
            <span>
              <button
                class="cf-push-right usa-button-disabled usa-button"
                disabled=""
                id="Add-contractor-button-id-1"
                type="button"
              >
                Add contractor
              </button>
            </span>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      aria-describedby="modal_id-desc"
      aria-labelledby="modal_id-title"
      aria-modal="true"
      class="cf-modal active "
      id="modal_id"
      role="dialog"
    >
      <div
        class="cf-modal-body"
        id="custom-contractor-modal"
      >
        <button
          class="cf-modal-close"
          id="Add-contractor-button-id-close"
          type="button"
        >
          <span
            class="usa-sr-only"
          >
            Close
          </span>
          <svg
            class="cf-icon-close"
            height="55"
            viewBox="0 0 55 55"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              close
            </title>
            <g
              fill="#212121"
            >
              <path
                d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
              />
            </g>
          </svg>
        </button>
        <div
          style="display: flex;"
        >
          <div
            data-css-1gs0ko2=""
          >
            <h1
              id="modal_id-title"
            >
              Add contractor
            </h1>
            <div
              data-css-o98ubo=""
            >
              <p>
                This will add a contractor to the list of assignable contractors.
              </p>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="name"
                >
                  <span>
                    Contractor name
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="name"
                    name="name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="directory"
                >
                  <span>
                    Contractor folder name in box.com
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="directory"
                    name="directory"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="poc"
                >
                  <span>
                    POC name
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="poc"
                    name="poc"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="phone"
                >
                  <span>
                    POC phone number
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="phone"
                    name="phone"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="cf-form-textinput   "
              >
                <label
                  for="email"
                >
                  <span>
                    POC email
                  </span>
                </label>
                <div
                  class="input-container"
                >
                  <input
                    class="cf-form-textinput"
                    id="email"
                    name="email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="cf-modal-divider"
        />
        <div
          class="cf-modal-controls"
        >
          <span>
            <button
              class="cf-modal-link cf-btn-link cf-push-left usa-button"
              id="Add-contractor-button-id-0"
              type="button"
            >
              Cancel
            </button>
          </span>
          <span>
            <button
              class="cf-push-right usa-button-disabled usa-button"
              disabled=""
              id="Add-contractor-button-id-1"
              type="button"
            >
              Add contractor
            </button>
          </span>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
